pkg/awsutils/awsutils.go

```diff
@@ -603,89 +603,130 @@ func (cache *EC2InstanceMetadataCache) getENIMetadata(eniMAC string) (ENIMetadat
 	}
 
 	log.Debugf("Found ENI: %s, MAC %s, device %d", eniID, eniMAC, deviceNum)
-
-	// Get IPv4 and IPv6 addresses assigned to interface
-	cidr, err := cache.imds.GetSubnetIPv4CIDRBlock(ctx, eniMAC)
-	if err != nil {
-		awsAPIErrInc("GetSubnetIPv4CIDRBlock", err)
-		return ENIMetadata{}, err
-	}
-
-	imdsIPv4s, err := cache.imds.GetLocalIPv4s(ctx, eniMAC)
+	// Get IMDS fields for the interface
+	macImdsFields, err := cache.imds.GetMACImdsFields(ctx, eniMAC)
 	if err != nil {
-		awsAPIErrInc("GetLocalIPv4s", err)
+		awsAPIErrInc("GetMACImdsFields", err)
 		return ENIMetadata{}, err
 	}
-
-	ec2ip4s := make([]*ec2.NetworkInterfacePrivateIpAddress, len(imdsIPv4s))
-	for i, ip4 := range imdsIPv4s {
-		ec2ip4s[i] = &ec2.NetworkInterfacePrivateIpAddress{
-			Primary:          aws.Bool(i == 0),
-			PrivateIpAddress: aws.String(ip4.String()),
+	isEFAOnlyInterface := true
```

Comment:
Why don't we flip the logic here?
```
	isEFAOnlyInterface := false
```
And if it true, then immediately return with appropriate empty fields

```
return ENIMetadata{
		ENIID:          eniID,
		MAC:            eniMAC,
		DeviceNumber:   deviceNum,
		SubnetIPv4CIDR: [],
		IPv4Addresses:  [],
		IPv4Prefixes:   [],
		SubnetIPv6CIDR: [],
```


Rest look good to me.

Reply:
Yeah we can do that

Reply:
Updated